From 58842a113a66ab10c1ed1c1d9d0c5158a605995f Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Tue, 27 Feb 2018 15:44:50 -0800
Subject: [PATCH] TRITON-189 change in TRITON-184 broke config-agent running in
 zones Reviewed by: Cody Peter Mello <cody.mello@joyent.com> Approved by: Cody
 Peter Mello <cody.mello@joyent.com>

---
 agent.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/agent.js b/agent.js
index 9f94266..f07a1fe 100644
--- a/agent.js
+++ b/agent.js
@@ -135,7 +135,8 @@ async.waterfall([
 				// Allow the config file to specify the
 				// instance UUID(s). This is used for load
 				// testing.
-				if (config.instances.length === 0) {
+				if (!config.instances ||
+				    config.instances.length === 0) {
 					config.instances = [ zonename ];
 				}
 			}
-- 
2.21.0

